setup-local/src/binaryControl.js

```diff
@@ -168,26 +194,39 @@ class BinaryControl {
    * Starts Local Binary using the args generated for this action
    */
   async startBinary() {
-    try {
-      this._generateArgsForBinary();
-      let { localIdentifier } = this.stateForBinary;
-      localIdentifier = localIdentifier ? `with local-identifier=${localIdentifier}` : '';
-      core.info(`Starting local tunnel ${localIdentifier} in daemon mode...`);
+    let { localIdentifier } = this.stateForBinary;
+    localIdentifier = localIdentifier ? `with local-identifier=${localIdentifier}` : '';
+    core.info(`Starting local tunnel ${localIdentifier} in daemon mode...`);
 
-      const { output, error } = await this._triggerBinary(LOCAL_TESTING.START);
+    let triesAvailable = BINARY_MAX_TRIES;
+
+    while (triesAvailable--) {
+      try {
+        this._generateArgsForBinary();
```

Comment:
any specific reason to move this inside `while`?

Reply:
let me try to refactor...

Reply:
done

Reply:
resolving it